src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/builder/SpecificGrantTypeQueryBuilder.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.hmcts.ccd.data.casedetails.search.builder;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.GrantType;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+
+@Slf4j
+@Component
+public class SpecificGrantTypeQueryBuilder implements GrantTypeQueryBuilder {
+
+    @Override
+    public String createQuery(List<RoleAssignment> roleAssignments, Map<String, Object> params) {
+        Supplier<Stream<RoleAssignment>> streamSupplier = () -> roleAssignments.stream()
+            .filter(roleAssignment -> GrantType.SPECIFIC.name().equals(roleAssignment.getGrantType()))
+            .filter(roleAssignment -> roleAssignment.getAuthorisations() == null
+                || roleAssignment.getAuthorisations().size() == 0);
+
+        String classficationQuery = createClassification(params, streamSupplier.get());
+
+        String tmpQuery = classficationQuery;
+
+        Set<String> jurisdictions = streamSupplier.get()
+            .map(roleAssignment -> roleAssignment.getAttributes().getJurisdiction().orElse(""))
+            .filter(jurisdiction -> jurisdiction.length() > 0)
+            .collect(Collectors.toSet());
+
+        if (jurisdictions.size() > 0) {
+            params.put("jurisdictions", jurisdictions);
+            tmpQuery = tmpQuery +  getOperator(tmpQuery, " AND ") + JURISDICTION + " in (:jurisdictions)";
+        }
+
+        Set<String> caseIds = streamSupplier.get()
```

Comment:
```suggestion
        Set<String> references = streamSupplier.get()
```
Best to name the variable `references` to avoid confusion, as the "`case_id`" in role assignments maps to the `reference` as we call it in CCD.

Reply:
changed to case References